File: P_4/Examen_P4/Napakalaki_Ruby_P4/lib/treasure.rb

module NapakalakiRuby
    
require_relative 'treasure_kind'
   
class Treasure
    
    #Consultores basicos
    attr_reader :name
    attr_reader :bonus
    attr_reader :type
    
    # Constructor
    def initialize(name = "", bonus = 0, type = Array.new)
        @name = name
        @bonus = bonus
        @type = type
    end
    
    # Metodos
    def to_s
        "Nombre: #{@name}
         Bonus: #{@bonus.to_s}
         Tipo: #{@type.to_s}"
    end
end

end

File: P_5/Napakalaki_Java_P5/src/napakalaki/Treasure.java

package napakalaki;

/**
 *
 * @author ixjosemi
 */
public class Treasure {

    private String name;
    private int bonus;
    private TreasureKind type;

    // Constructor
    public Treasure (String n, int bonus, TreasureKind t){
        name = n;
        bonus = bonus;
        type = t;
    }
    
    // Devuelve el nombre del tesoro
    public String getName(){
        return name;
    }
    
    // Devuelve el bonus que aporta el tesoro
    public int getBonus(){
        return bonus;
    }
    
    // Devuelve el tipo de tesoro
    public TreasureKind getType(){
        return type;
    }

    @Override
    public String toString() {
        return   "Treasure: " + name 
                + "\n bonus = " + bonus 
                + "\n type = " + type;
    }
    
    
}
